mantidproject/mantid
# Update MVP tutorial documentation

### Description of work
This PR updates some of the MVP tutorial documentation to use some coding best practises in the examples, and also adds MVP best practises and an explanation why they are useful.

It also simplifies some of the examples by removing the use of the QMainWindow. Instead it is useful to just use a plain QWidget because it simplifies code, but also it means the QWidget you design can be used standalone OR in embedded within a wider interface. (QMainWindow is less flexible as it cannot be embedded in a wider interface).

Part of #37830
Still need to update the MVPDesign document in a separate PR

### To test:
Follow the MVP tutorial
Make sure you an open the example Calculator widget, and the Final widget at the end of the tutorial

*This does not require release notes* because **it is an internal documentation change**


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
